feeadjuster/feeadjuster.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+import random
+import time
+from pyln.client import Plugin, Millisatoshi, RpcError
+
+
+plugin = Plugin()
+# Our amount and the total amount in each of our channel, indexed by scid
+plugin.adj_balances = {}
+# Cache to avoid loads of calls to getinfo
+plugin.our_node_id = None
+# Users can configure this
+plugin.update_treshold = 0.05
+
+
+def get_ratio(our_percentage):
+    """
+    Basic algorithm: the farther we are from the optimal case, the more we
+    bump/lower.
+    """
+    return 50**(0.5 - our_percentage)
+
+
+def maybe_adjust_fees(plugin: Plugin, scids: list):
+    for scid in scids:
+        our = plugin.adj_balances[scid]["our"]
+        total = plugin.adj_balances[scid]["total"]
+        percentage = our / total
+        last_percentage = plugin.adj_balances[scid].get("last_percentage")
+
+        # Only update on substantial balance moves to avoid flooding, and add
+        # some pseudo-randomness to avoid too easy channel balance probing
+        update_treshold = plugin.update_treshold
+        if not plugin.deactivate_fuzz:
+            update_treshold += random.uniform(-0.015, 0.015)
+
+        if (last_percentage is None
+                or abs(last_percentage - percentage) > update_treshold):
+            ratio = get_ratio(percentage)
+            try:
+                plugin.rpc.setchannelfee(scid, int(plugin.adj_basefee * ratio),
+                                         int(plugin.adj_ppmfee * ratio))
```

Comment:
This is where the range of the ratio function is important. As is this allows only an adjustment to `[1/7;7]` times the normal fee.

Reply:
I wanted to be conservative with variations by default, but really happy to bikeshed my tinkered formula :)